src/Datadog.Trace/Tagging/AzureFunctionsTags.cs

```diff
@@ -0,0 +1,50 @@
+// <copyright file="AzureFunctionsTags.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System.Collections.Generic;
+using Datadog.Trace.ExtensionMethods;
+
+namespace Datadog.Trace.Tagging
+{
+    internal class AzureFunctionsTags : InstrumentationTags
+    {
+        public static readonly IEnumerable<Property<AzureFunctionsTags, string>> AzureFunctionsExtraTags = new List<Property<AzureFunctionsTags, string>>()
+        {
+            new ReadOnlyProperty<AzureFunctionsTags, string>(Trace.Tags.InstrumentationName, t => t.InstrumentationName),
+            new Property<AzureFunctionsTags, string>(Trace.Tags.AzureFunctionTriggerType, t => t.TriggerType, (t, v) => t.TriggerType = v),
+            new Property<AzureFunctionsTags, string>(Trace.Tags.AzureFunctionName, t => t.ShortName, (t, v) => t.ShortName = v),
+            new Property<AzureFunctionsTags, string>(Trace.Tags.AzureFunctionMethod, t => t.FullName, (t, v) => t.FullName = v),
+            new Property<AzureFunctionsTags, string>(Trace.Tags.AzureFunctionBindingSource, t => t.BindingSource, (t, v) => t.BindingSource = v)
+        };
+
+        public static readonly IProperty<string>[] AzureFunctionRootTagsProperties =
+            InstrumentationTagsProperties.Concat(
```

Comment:
I think you can save yourself from creating everything a second time and just calling `InstrumentationTagsProperties.Concat(AzureFunctionsExtraTags)`. It'd be nice to only list them once so we have one source of truth on the set of tags

Reply:
Generic constraints did not allow this, as you'll notice they are different kinds of collections.
I would have to use Linq to convert these.
I looked at creating a function that returned these and it didn't seem better to me, but I can look again.